### PR TITLE
net/arpwatch:  Bring back, update, and improve arpwatch

### PR DESCRIPTION
**Pull request by dfc-d**
Arpwatch detects new stations (mac address associated
with a given ip), changed stations, and stations with
ip on a subnet other than the one on the interface on
which the station was detected (bogon).  It can log
and/or mail detections using any mailer with a sendmail
compatible command.

I've created the package assuming nullmailer will be used (i.e. adding arpwatch to the nullmailer group if it exists in order for nullmailer's sendmail command to succeed).  This has no effect if nullmailer is not installed.

Signed-off-by: Daniel Dickinson <openwrt@daniel.thecshore.com>

### CODE REVIEW

**Comment by ryzhovau** [comment]
Is there any reason to pack empty `arp.dat` file? May be better to create it on run time, if necessary.

Also it's a data base, not sure it must be placed to `/etc` and marked as config. Debian guys put it to `/var/lib/arpwatch`.

**Comment by dfc-d** [comment]
The arp.dat was because I though I had to have a physical file to make it a conffile.

Making it a conffile *is* necessary because unlike debian an upgrade means everything except what is defined as a conffile get erased and only 'conffiles' are kept across and upgrade.

Another difference from debian is that /var is a symlink to /tmp and goes away on every boot.

In the case of arpwatch arp.dat keeps track of what stations have been seen before and if it goes away on boot or upgrade you will get a flood of new station notifications as all your existing stations get detected as new stations.

In short you can't really use debian as a meaningful guidepost as to what should happen in openwrt (the network is also done completely differently, and uci based config is preferred to using the stock config files in /etc).

Please don't try to impose debian ideas on openwrt - it won't work.

**Comment by dfc-d** [comment]
I've eliminated shipping arp.dat but otherwise I believe the package is correct and the way it should be.

**Comment by dfc-d** [comment]
BSD-4-Clause is non-OSI and makes this software mostly irrelevant.
